src/storages/sled_storage/auto_increment.rs

```diff
@@ -0,0 +1,81 @@
+#![cfg(feature = "auto-increment")]
+use {
+    super::{error::err_into, SledStorage},
+    crate::{AutoIncrement, MutResult, Row, Value},
+    async_trait::async_trait,
+    fstrings::*,
+    sqlparser::ast::ColumnDef,
+};
+
+macro_rules! try_into {
+    ($self: expr, $expr: expr) => {
+        match $expr {
+            Err(e) => {
+                return Err(($self, e));
+            }
+            Ok(v) => v,
+        }
+    };
+}
+
+#[async_trait(?Send)]
+impl AutoIncrement for SledStorage {
+    async fn generate_values(
+        self,
+        table_name: &str,
+        columns: Vec<(usize, &ColumnDef)>,
+        rows: Vec<Row>,
+    ) -> MutResult<Self, Vec<Row>> {
+        // FAIL: No-mut
+        let mut storage = self;
+        let mut rows = rows;
+        for column in columns.iter() {
+            let result = generate_column_values(storage, table_name, column, rows).await?;
+            storage = result.0;
+            rows = result.1;
+        }
+        Ok((storage, rows))
```

Comment:
fyi. here you can use :)
```rust
use futures::stream::{self, TryStreamExt};                             
                                                                       
stream::iter(columns.iter().map(Ok))                                   
    .try_fold((self, rows), |(storage, rows), column| async move {     
        generate_column_values(storage, table_name, column, rows).await
    })                                                                 
    .await                                                             
```

Reply:
Cheater!
I couldn't get that to work hahaha! Thanks.

Reply:
Looks like it works. Thanks.

Reply:
I think I wrote nearly that exact code a few times, just missing the `move` and I think I mightn't have had the `use TryStreamExt`.

Reply:
Yeah, both are important.
`try_fold` is in `TryStreamExt` so it is necessary to be imported. :)

Because async requires ownership of values, so `move` should be there for ensuring the values inside are moved, not borrowed.

fyi. https://github.com/rust-lang/rust/issues/62290
The reason we use `async move { .. }` block statement inside closure is current stable rust does not support async closure.
I'm really looking forward to see async closure in stable rust.. :)